commit 4dc4abfdc77e7fe01edd4b94005b55cf8c8a8954 (refs/changes/88/3788/2)
Author: Robert Bogart <robert.bogart@joyent.com>
Date:   2018-04-10T17:11:39+00:00 (1 year, 6 months ago)
    
    MANTA-3598 electric-moray artedi metrics could provide similar fields to moray's
    Reviewed by: Jan Wyszynski <jan.wyszynski@joyent.com>
    Reviewed by: Kody A Kantor <kody.kantor@joyent.com>
    Approved by: Jan Wyszynski <jan.wyszynski@joyent.com>
    Approved by: Kody A Kantor <kody.kantor@joyent.com>

diff --git a/lib/server.js b/lib/server.js
index fbd7f40..7f6d12c 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -155,7 +155,16 @@ function createServer(options, callback) {
             opts.clients = clients;
             opts.indexShards = options.ringCfg.indexShards;
 
-            var collector = artedi.createCollector();
+            var labels = {
+                datacenter: options.datacenter,
+                server: options.server_uuid,
+                zonename: os.hostname(),
+                pid: process.pid
+            };
+
+            var collector = artedi.createCollector({
+                labels: labels
+            });
 
             var socket = net.createServer({ 'allowHalfOpen': true });
             var server = new fast.FastServer({
diff --git a/sapi_manifests/electric-moray/template b/sapi_manifests/electric-moray/template
index a3d0d9b..b8cac1d 100644
--- a/sapi_manifests/electric-moray/template
+++ b/sapi_manifests/electric-moray/template
@@ -40,5 +40,7 @@
                 "readOnly": {{readOnly}}{{/readOnly}}
             }{{^last}},{{/last}}{{/INDEX_MORAY_SHARDS}}
         ]
-    }
+    },
+    "server_uuid": "{{auto.SERVER_UUID}}",
+    "datacenter": "{{DATACENTER}}"
 }
